### PR TITLE
[beta] Add function to download a vector layer

### PR DESCRIPTION
**Pull request by ao-picterra**

### CODE REVIEW

File: src/picterra/client.py [hunk 1]
```diff
@@ -943,6 +943,32 @@ def delete_vector_layer(self, vector_layer_id: UUID):
...
             raise APIError(resp.text)
 
+    def download_vector_layer_to_file(self, vector_layer_id: UUID, filename: str):
+        """
+        Downloads a vector layer
+
+        This a **beta** function, subject to change.
+
+        Args:
+            vector_layer_id: The id of the vector layer to remove
```
Comment (romankaravia) [inline src/picterra/client.py:953]: left-over from copy-pasting? We are not removing anything here, right?

File: src/picterra/client.py [hunk 2]
```diff
@@ -943,6 +943,32 @@ def delete_vector_layer(self, vector_layer_id: UUID):
...
+            filename: existing file to save the vector layer in
+        """
+        if not (os.path.exists(filename) and os.path.isfile(filename)):
```
Comment (romankaravia) [inline src/picterra/client.py:956]: It's also documented like this 2 lines above, but why does the file need to exist already? Couldn't it just be created if it doesn't exist yet?

**Review by romankaravia — COMMENTED** [review]
Looks good otherwise, maybe I'm just misunderstanding something about the file existence check??

File: src/picterra/client.py (hunk 1 above)
Comment (ao-picterra) [reply src/picterra/client.py:953]: fixed

File: src/picterra/client.py (hunk 2 above)
Comment (ao-picterra) [reply src/picterra/client.py:956]: fixed

**Pull request merged**